### PR TITLE
changed grafana builder to cuisine style

### PR DESCRIPTION
**Pull request by mie00**

### CODE REVIEW

File: lib/JumpScale/tools/cuisine_/CuisineBuilder.py [hunk 1]
```diff
@@ -933,27 +942,29 @@ def grafana(self, start=True, influx_addr='127.0.0.1', influx_port=8086):
...
-              'url': 'http://%s:%u' % (influx_addr, influx_port),
-              'user': 'admin',
-              'password': 'passwd',
-              'default': True,
-            }
-            grafanaclient.addDataSource(data)
+            self._grafanaStart(influx_addr=influx_addr, influx_port=influx_port)
+
+    def _startGrafana(self, influx_addr='127.0.0.1', influx_port=8086):
+        j.tools.cuisine.local.file_copy('$tmplsDir/grafana/conf/defaults.ini', '$cfgDir/grafana.ini')
```
Comment (zaibon) [inline lib/JumpScale/tools/cuisine_/CuisineBuilder.py]: should not use local cuisine.

File: lib/JumpScale/tools/cuisine_/CuisineBuilder.py [hunk 2]
```diff
@@ -933,27 +942,29 @@ def grafana(self, start=True, influx_addr='127.0.0.1', influx_port=8086):
...
+        cmd = "$binDir/grafana-server --config=$cfgDir/grafana.ini"
```
Comment (zaibon) [inline lib/JumpScale/tools/cuisine_/CuisineBuilder.py]: same remark about configuration location

File: lib/JumpScale/tools/cuisine_/CuisineBuilder.py [hunk 3]
```diff
@@ -933,27 +942,29 @@ def grafana(self, start=True, influx_addr='127.0.0.1', influx_port=8086):
...
+        self.cuisine.process.kill("grafana-server")
+        self.cuisine.processmanager.ensure("grafana-server", cmd=cmd, env={}, path="$tmplsDir/grafana/")
+        import time
+        time.sleep(10)
+        grafanaclient = j.clients.grafana.get(url='http://%s:3000'%(self.cuisine.executor.addr),username='admin', password='admin')
```
Comment (zaibon) [inline lib/JumpScale/tools/cuisine_/CuisineBuilder.py]: no possibility to choose on which port grafana listen to ?

File: lib/JumpScale/tools/cuisine_/CuisineBuilder.py [hunk 4]
```diff
@@ -766,19 +766,28 @@ def influxdb(self, start=True):
 
     def grafana(self, start=True, influx_addr='127.0.0.1', influx_port=8086):
 
-        if self.cuisine.isMac:
-            self.cuisine.package.mdupdate()
-            self.cuisine.package.install('grafana')
         if self.cuisine.isUbuntu:
+            dataDir = self.cuisine.args_replace("$varDir/grafana/data")
```
Comment (zaibon) [inline lib/JumpScale/tools/cuisine_/CuisineBuilder.py]: convention for data is `$varDir/data/$app_name` so should be `$varDir/data/grafana` instead

File: lib/JumpScale/tools/cuisine_/CuisineBuilder.py [hunk 5]
```diff
@@ -933,27 +942,29 @@ def grafana(self, start=True, influx_addr='127.0.0.1', influx_port=8086):
...
-              'default': True,
-            }
-            grafanaclient.addDataSource(data)
+            self._grafanaStart(influx_addr=influx_addr, influx_port=influx_port)
+
+    def _startGrafana(self, influx_addr='127.0.0.1', influx_port=8086):
+        j.tools.cuisine.local.file_copy('$tmplsDir/grafana/conf/defaults.ini', '$cfgDir/grafana.ini')
+        cmd = "$binDir/grafana-server --config=$cfgDir/grafana.ini"
+        self.cuisine.process.kill("grafana-server")
+        self.cuisine.processmanager.ensure("grafana-server", cmd=cmd, env={}, path="$tmplsDir/grafana/")
```
Comment (zaibon) [inline lib/JumpScale/tools/cuisine_/CuisineBuilder.py]: I think the running server should not read from the template directory. Can't we copy the template in $cfgDir and read from there ?

File: lib/JumpScale/tools/cuisine_/CuisineBuilder.py (hunk 3 above)
Comment (mie00) [inline lib/JumpScale/tools/cuisine_/CuisineBuilder.py]: No it's possible. should we make that an argument to start?

Comment (zaibon) [inline lib/JumpScale/tools/cuisine_/CuisineBuilder.py]: Yes please

File: lib/JumpScale/tools/cuisine_/CuisineBuilder.py (hunk 5 above)
Comment (mie00) [inline lib/JumpScale/tools/cuisine_/CuisineBuilder.py]: Yes we can, but These files are not configuration. they are static files to be served by grafana.

Comment (zaibon) [inline lib/JumpScale/tools/cuisine_/CuisineBuilder.py]: ok, leave it then

**Pull request merged**